core/src/main/java/org/wso2/carbon/kernel/internal/runtime/RuntimeServiceListenerComponent.java

```diff
@@ -70,6 +71,7 @@ protected void start(BundleContext bundleContext) {
     protected void registerRuntime(Runtime runtime) {
         try {
             runtimeManager.registerRuntime(runtime);
+            StartupServiceUtils.updateServiceCache("carbon-runtime-mgt", Runtime.class, runtime);
```

Comment:
Updated the component names to constants via 00640b0b427de47168b56103e30a5709b419cdb1